### PR TITLE
feat: add a xmpp server to full tls docker-compose

### PR DESCRIPTION
**Pull request by douglaz**
As fedimint uses ecash, which works very well through chat, it may be useful for test federations to have a standardized chat server like xmpp.

Uses prosody, which is very stable and lightweight. Tested on one machine with 2GB of RAM and there is no increase in memory or cpu requirements compared to current version.

### CODE REVIEW

File: docker/full-tls-mutinynet/docker-compose.yaml [hunk 1]
```diff
@@ -241,6 +241,172 @@ services:
     restart: always
     platform: linux/amd64
 
+  prosody_config_writer:
```
Comment (otech47) [inline docker/full-tls-mutinynet/docker-compose.yaml:244]: cant all instances of `fedimint.my-super-host.com` be written in as an environment variable or no?

Comment (douglaz) [reply docker/full-tls-mutinynet/docker-compose.yaml:244]: Not sure :thinking:

**Review by otech47 — COMMENTED** [review]
I just tried testing this with a [tweaked commit](https://github.com/setlife-network/fedimint/commit/36a29121a0a2b487cce725d90aaeadb0962cba39) using my own DNS and get a failure to download the image

Screenshot 2023-10-25 at 11 10 19 AM

I guess because this PR needs to be merged first for the image to be deployed?

on a broader note I remember the install script used to prompt you to install the gateway `y/n` but i don't think i got that prompt anymore... i was going to suggest that this xmpp server also be an optional install. what do you think @douglaz? or does the same reasoning for removing the gateway optional install apply here as well?

**Comment by douglaz** [comment]
> concept ACK
> 
> I just tried testing this with a [tweaked commit](https://github.com/setlife-network/fedimint/commit/36a29121a0a2b487cce725d90aaeadb0962cba39) using my own DNS and get a failure to download the image
> 
> Screenshot 2023-10-25 at 11 10 19 AM
> 
> I guess because this PR needs to be merged first for the image to be deployed?
> 
> on a broader note I remember the install script used to prompt you to install the gateway `y/n` but i don't think i got that prompt anymore... i was going to suggest that this xmpp server also be an optional install. what do you think @douglaz? or does the same reasoning for removing the gateway optional install apply here as well?

Yeah, it needs to be merged first. The prompt is defined only for the non-tls version.

**Review by justinmoon — APPROVED** [review]

File: docker/full-tls-mutinynet/docker-compose.yaml [hunk 2]
```diff
@@ -44,6 +43,7 @@ services:
       - FM_P2P_URL=fedimint://fedimint.my-super-host.com:8173
       - FM_BIND_API=0.0.0.0:80
       - FM_API_URL=wss://fedimintd.fedimint.my-super-host.com
+      - FM_EXTRA_DKG_META=chat_server_domain=xmpp.fedimint.my-super-host.com
```
Comment (elsirion) [inline docker/full-tls-mutinynet/docker-compose.yaml:46]: The chat server meta field needs to be standardized by adding a description in [`docs/meta_fields`](https://github.com/fedimint/fedimint/tree/master/docs/meta_fields) or a similar external repository (in that case please prefix with the organization name).

**Pull request merged**